javav2/example_code/iotsitewise/src/main/java/com/example/iotsitewise/scenario/SitewiseActions.java

```diff
@@ -0,0 +1,479 @@
+                    .entryId("entry-3")
+                    .assetId(assetId)
+                    .propertyId(tempPropertyId)
+                    .propertyValues(Arrays.asList(
+                        AssetPropertyValue.builder()
+                            .value(Variant.builder()
+                                .doubleValue(sampleData.get("Temperature"))
+                                .build())
+                            .timestamp(time)
+                            .build()
+                    ))
+                    .build(),
+                PutAssetPropertyValueEntry.builder()
+                    .entryId("entry-4")
+                    .assetId(assetId)
+                    .propertyId(humidityPropId)
+                    .propertyValues(Arrays.asList(
+                        AssetPropertyValue.builder()
+                            .value(Variant.builder()
+                                .doubleValue(sampleData.get("Humidity"))
+                                .build())
+                            .timestamp(time)
+                            .build()
+                    ))
+                    .build()
+            ))
+            .build();
+
+        return getAsyncClient().batchPutAssetPropertyValue(request)
+            .whenComplete((response, exception) -> {
+                if (exception != null) {
+                    logger.error("An exception occurred: {}", exception.getCause().getMessage());
+                } else {
+                    logger.info("Data sent successfully.");
+                }
+            });
+    }
+    // snippet-end:[sitewise.java2_put_batch_property.main]
+
+    // snippet-start:[sitewise.java2_get_property.main]
+    // TODO -- fix this including Javadoc
+    public void getAssetPropValueAsync(String propName, String propId, String assetId) {
+        GetAssetPropertyValueRequest assetPropertyValueRequest = GetAssetPropertyValueRequest.builder()
+            .propertyId(propId)
+            .assetId(assetId)
+            .build();
+
+        CompletableFuture<GetAssetPropertyValueResponse> futureResponse = getAsyncClient().getAssetPropertyValue(assetPropertyValueRequest);
+        futureResponse.whenComplete((response, exception) -> {
+            if (exception != null) {
+                throw new RuntimeException("Error occurred while fetching property value: " + exception.getMessage(), exception);
+            } else {
+                String assetVal = String.valueOf(response.propertyValue().value().doubleValue());
+                logger.info("The property name is: {} ", propName);
+                logger.info("The value of this property is {}", assetVal);
+            }
+        }).join();
+    }
+    // snippet-end:[sitewise.java2_get_property.main]
+
+    // snippet-start:[sitewise.java2.describe.asset.model.main]
+    /**
+     * @param assetModelId The Id of the asset model.
+     * @return A map of the asset model properties when the CompletableFuture completes.
+     */
```

Comment:
It returns and CompletableFuture that ...

Reply:
updated this.